model/src/lib.rs

```diff
@@ -71,6 +72,18 @@ impl Crd {
             Self::Resource(resource) => resource.metadata.name.to_owned(),
         }
     }
+
+    pub fn labels(&self) -> BTreeMap<String, String> {
+        match self {
+            Self::Test(test) => test.metadata.labels.as_ref().cloned().unwrap_or_default(),
+            Self::Resource(resource) => resource
+                .metadata
+                .labels
+                .as_ref()
+                .cloned()
```

Comment:
Can `as_ref().clone()` be simplified to  `to_owned()`?
```suggestion
            Self::Test(test) => test.metadata.labels.to_owned().unwrap_or_default(),
            Self::Resource(resource) => resource
                .metadata
                .labels
                .to_owned()
```